app/components/views/DexPage/index.js

```diff
@@ -0,0 +1,78 @@
+import { RegisterPageContent, RegisterPageHeader } from "./RegisterPage";
+import { DexViewContent, DexViewHeader } from "./DexView";
+import { useDex } from "./hooks";
+import { StandalonePage, StandaloneHeader } from "layout";
+import {
+  CreateWalletPageContent,
+  CreateWalletPageHeader
+} from "./CreateWalletPage";
+import { EnablePageContent, EnablePageHeader } from "./EnablePage";
+import { InitPageContent, InitPageHeader } from "./InitPage";
+import { LoginPageContent, LoginPageHeader } from "./LoginPage";
+import {
+  CreateDexAccountPageContent,
+  CreateDexAccountPageHeader
+} from "./CreateDexAccountPage";
+import { DEX_ICON } from "constants";
+import { FormattedMessage as T } from "react-intl";
+
+const DexPage = () => {
+  let Page, Header;
+  const {
+    dexActive,
+    dexInit,
+    loggedIn,
+    dexRegistered,
+    dexDCRWalletRunning,
+    dexEnabled,
+    dexBTCWalletRunning,
+    dexAccount
+  } = useDex();
+  if (!dexEnabled) {
```

Comment:
use `useMemo` to memorize Page & Header components.

Reply:
this also can be moved to `useDex` which can return `{ Page, Header }` component and encapsulate all the 
state logic.